packages/raja/package.py

```diff
@@ -55,22 +57,28 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     )
 
     variant("openmp", default=True, description="Build OpenMP backend")
-    variant("shared", default=True, description="Build Shared Libs")
+    variant("shared", default=True, description="Build shared libs")
     variant("examples", default=True, description="Build examples.")
     variant("exercises", default=True, description="Build exercises.")
     # TODO: figure out gtest dependency and then set this default True
     # and remove the +tests conflict below.
     variant("tests", default=False, description="Build tests")
     variant("libcpp", default=False, description="Uses libc++ instead of libstdc++")
-    variant("desul", default=False, description="Build Desul Atomics backend")
+    variant("desul", default=False, description="Build desul atomics backend")
+    variant("vectorization", default=True, description="Build SIMD/SIMT intrinsics support")
 
     depends_on("blt")
+# specific commit used here to be consistent with camp -- no matching BLT release
+    depends_on("blt@8c229991e65e7a9603c621b47cb3ba158bb7468c", type="build", when="@2022.10.3")
```

Comment:
@rhornung67 @davidbeckingsale I am going to merge this in #46 because we need to test those changes quickly.

However, I think we should clarify something about submodules and spack dependencies.

I understand that we want to download here the exact same version of BLT used as submodule in Camp. In its spack package Camp has BLT set as a dependency, with "submodules=True".

I think we want the synchronization of BLT across projects to be automatic. And I think we can achieve that by placing this line (with a specific commit of BLT) in the camp package, and set submodules to False.

In any case, I am not certain what Spack recommends for submodule usage. Maybe we should invite @becker33 ?

Reply:
@adrienbernede whatever you think is best. I'm still coming up to speed.

Reply:
Like Cyrus said, the Spack-provided BLT will be used regardless of the existence of the submodule: https://github.com/LLNL/radiuss-spack-configs/blob/woptim/gather-local-packages/packages/camp/package.py#L149

Reply:
@cyrush I seem to remember one can select the BLT modules do download.

Additionally to the confusion it creates, I’ve had issues that I connected to the use of submodules=true when the dependency is also fetch by Spack: admittedly, I did not go into the details there, maybe that would be worth the effort.

Reply:
FWIW, I think we should not be using submodules in our CI testing at all in Gitlab CI since none of our main users do that. We do run tests with the submodules in Azure, for example. I think that is sufficient. 

Also, I think the CMake in RAJA, Umpire, CHAI, camp, etc. has logic to avoid issues with different BLTs being used in an integrated build. So we should be safe.

Reply:
As long as submodules are tested. 

I am probably the odd one out, but I expect the submodules to work out of the box vs relying on spack to solve the version issue for me. We don't use spack for all deployments and I have never had to add a BLT specific checkout.

Reply:
@rhornung67 Final notes:
- While testing this in CI I had troubles. Discussing them with Greg I learned that git-hash-versions are actually not supported in Spack. So I dropped that...
- The way it was coded here, that specific version of blt would only have been used with raja@2022.10.3, not with develop. This was not tested in CI, explaining why you had no issue with it yourself.

Reply:
OK. Thanks for the clarification.